Device/Device.cs

```diff
@@ -47,23 +45,35 @@ public async Task ConnectDevice(string willPayload = "")
                 .WithCredentials(username, password)
                 .WithClientId(deviceId)
                 .WithProtocolVersion(MqttProtocolVersion.V311)
+                .WithCleanSession(false)
                 .WithTls()
                 .WithWillMessage(ConstructWillMessage(willPayload))
                 .Build();
+            try
+            {
+                await mqttClient.ConnectAsync(options, CancellationToken.None);
+            }
+            catch (TaskCanceledException)
+            {
+                // This is expected when the token is signaled; it should not be considered an

```

Comment:
Should we at least add a `Console.WriteLine(e.ToString())` in case something unexpected also triggers a `TaskCanceledException` ?

Reply:
Sure :)